Adjust rolling upgrade test parameters

### Description

We are having some flaky failures of the `DefaultRollingUpgradeTestIT`. The failure usually involves the servers starting in a partitioned, the test check waits for the default 45s until the containers are clustered, but then it times out. The merge view is installed a few milliseconds later. For example:

```
2026-03-16T20:01:37.4588099Z Caused by: java.util.concurrent.TimeoutException: After the await period 45 SECONDS the count down should be 0 and is 3
2026-03-16T20:01:37.4589109Z 	at org.infinispan.server.test.core.CountdownLatchLoggingConsumer.await(CountdownLatchLoggingConsumer.java:31)
2026-03-16T20:01:37.4590122Z 	at org.infinispan.server.test.core.ContainerInfinispanServerDriver.lambda$start$3(ContainerInfinispanServerDriver.java:175)
2026-03-16T20:01:37.4590983Z 	at org.infinispan.testing.Exceptions.unchecked(Exceptions.java:302)
2026-03-16T20:01:37.4591673Z 	... 84 more
2026-03-16T20:01:37.4591828Z 
2026-03-16T20:01:37.6225689Z 20:01:37,448 INFO  [o.i.s.t.c.r.RollingUpgradeHandler] Configurations for version 16.0.4 using org.infinispan:infinispan-server-tests:16.0.4:configurations
2026-03-16T20:01:37.9230027Z 20:01:37,894 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#0]STDOUT: 2026-03-16 20:01:37,890 INFO  [o.i.CLUSTER] ISPN000093: Received new, MERGED cluster view for channel org.infinispan.server.rollingupgrade.DefaultRollingUpgradeTestIT: MergeView::[bcdec1eb5676-52036(v=16.0.1)|2] (3) [bcdec1eb5676-52036(v=16.0.1), 5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], 2 subgroups: [5be8d89e1034-37469(v=16.0.1)|1] (2) [5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], [bcdec1eb5676-52036(v=16.0.1)|0] (1) [bcdec1eb5676-52036(v=16.0.1)]
2026-03-16T20:01:37.9243467Z 20:01:37,897 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#0]STDOUT: 2026-03-16 20:01:37,895 INFO  [o.i.CLUSTER] [Scope=5be8d89e1034-37469]ISPN100000: Node bcdec1eb5676-52036 joined the cluster
2026-03-16T20:01:37.9246331Z 20:01:37,900 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#0]STDOUT: 2026-03-16 20:01:37,897 INFO  [o.i.CLUSTER] [Scope=5be8d89e1034-37469]ISPN100000: Node bcdec1eb5676-52036 joined the cluster
2026-03-16T20:01:37.9250315Z 20:01:37,900 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#1]STDOUT: 2026-03-16 20:01:37,898 INFO  [o.i.CLUSTER] ISPN000093: Received new, MERGED cluster view for channel org.infinispan.server.rollingupgrade.DefaultRollingUpgradeTestIT: MergeView::[bcdec1eb5676-52036(v=16.0.1)|2] (3) [bcdec1eb5676-52036(v=16.0.1), 5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], 2 subgroups: [5be8d89e1034-37469(v=16.0.1)|1] (2) [5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], [bcdec1eb5676-52036(v=16.0.1)|0] (1) [bcdec1eb5676-52036(v=16.0.1)]
2026-03-16T20:01:37.9254071Z 20:01:37,903 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#1]STDOUT: 2026-03-16 20:01:37,903 INFO  [o.i.CLUSTER] [Scope=bcdec1eb5676-52036]ISPN100000: Node 5be8d89e1034-37469 joined the cluster
2026-03-16T20:01:37.9255872Z 20:01:37,911 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#1]STDOUT: 2026-03-16 20:01:37,911 INFO  [o.i.CLUSTER] [Scope=bcdec1eb5676-52036]ISPN100000: Node 8b50682b0c55-2127 joined the cluster
2026-03-16T20:01:37.9257596Z 20:01:37,913 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#1]STDOUT: 2026-03-16 20:01:37,912 INFO  [o.i.CLUSTER] [Scope=bcdec1eb5676-52036]ISPN100000: Node 5be8d89e1034-37469 joined the cluster
2026-03-16T20:01:37.9259325Z 20:01:37,916 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#1]STDOUT: 2026-03-16 20:01:37,914 INFO  [o.i.CLUSTER] [Scope=bcdec1eb5676-52036]ISPN100000: Node 8b50682b0c55-2127 joined the cluster
2026-03-16T20:01:38.0324102Z 20:01:37,927 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#2]STDOUT: 2026-03-16 20:01:37,917 INFO  [o.i.CLUSTER] ISPN000093: Received new, MERGED cluster view for channel org.infinispan.server.rollingupgrade.DefaultRollingUpgradeTestIT: MergeView::[bcdec1eb5676-52036(v=16.0.1)|2] (3) [bcdec1eb5676-52036(v=16.0.1), 5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], 2 subgroups: [5be8d89e1034-37469(v=16.0.1)|1] (2) [5be8d89e1034-37469(v=16.0.1), 8b50682b0c55-2127(v=16.0.1)], [bcdec1eb5676-52036(v=16.0.1)|0] (1) [bcdec1eb5676-52036(v=16.0.1)]
2026-03-16T20:01:38.0331837Z 20:01:37,935 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#2]STDOUT: 2026-03-16 20:01:37,932 INFO  [o.i.CLUSTER] [Scope=8b50682b0c55-2127]ISPN100000: Node bcdec1eb5676-52036 joined the cluster
2026-03-16T20:01:38.0352826Z 20:01:37,939 INFO  [o.i.CONTAINER] [o.i.s.r.D.1.0.1#2]STDOUT: 2026-03-16 20:01:37,938 INFO  [o.i.CLUSTER] [Scope=8b50682b0c55-2127]ISPN100000: Node bcdec1eb5676-52036 joined the cluster
```

But then it's too late. The test has already failed.

### Implementation ideas

Adjust the starting sequence of containers, slightly increase the 45s timeout. 50s would be enough. Many other tests utilize a custom JGroups stack with the default values of MERGE3 updated. The `DefaultRollingUpgradeTestIT` runs with the default TCP stack.

--- a/server/rollingupgradetests/src/test/java/org/infinispan/server/rollingupgrade/DefaultRollingUpgradeTestIT.java
+++ b/server/rollingupgradetests/src/test/java/org/infinispan/server/rollingupgrade/DefaultRollingUpgradeTestIT.java
@@ -38,7 +38,10 @@ public class DefaultRollingUpgradeTestIT {
    public void testDefaultSetting() {
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder(DefaultRollingUpgradeTestIT.class.getName(),
             RollingUpgradeTestUtil.getFromVersion(), RollingUpgradeTestUtil.getToVersion());
-      RollingUpgradeConfiguration configuration = builder.jgroupsProtocol("tcp").build();
+      RollingUpgradeConfiguration configuration = builder
+            .useCustomServerConfiguration("configuration/ClusteredServerTest.xml")
+            .jgroupsProtocol("test-tcp")
+            .build();
       Assumptions.assumeFalse(Compatibility.INSTANCE.isCompatibilitySkip(configuration), "Incompatible test");
       RollingUpgradeHandler.performUpgrade(configuration);
    }
@@ -50,7 +53,8 @@ public class DefaultRollingUpgradeTestIT {
       int nodeCount = 3;
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder(DefaultRollingUpgradeTestIT.class.getName(),
             RollingUpgradeTestUtil.getFromVersion(), RollingUpgradeTestUtil.getToVersion())
-            .jgroupsProtocol("tcp")
+            .useCustomServerConfiguration("configuration/ClusteredServerTest.xml")
+            .jgroupsProtocol("test-tcp")
             .nodeCount(nodeCount);
       RestClientConfigurationBuilder restBuilder = new RestClientConfigurationBuilder();
       restBuilder.security().authentication().enable().username(user.getUser()).password(user.getPassword());
@@ -102,7 +106,8 @@ public class DefaultRollingUpgradeTestIT {
       ByRef.Integer interactions = new ByRef.Integer(0);
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder(DefaultRollingUpgradeTestIT.class.getName(),
             RollingUpgradeTestUtil.getFromVersion(), RollingUpgradeTestUtil.getToVersion())
-            .jgroupsProtocol("tcp")
+            .useCustomServerConfiguration("configuration/ClusteredServerTest.xml")
+            .jgroupsProtocol("test-tcp")
             .nodeCount(nodeCount);
 
       RedisURI.Builder respBuilder = RedisURI.builder()
@@ -156,7 +161,8 @@ public class DefaultRollingUpgradeTestIT {
 
       RollingUpgradeConfigurationBuilder builder = new RollingUpgradeConfigurationBuilder(DefaultRollingUpgradeTestIT.class.getName(),
             RollingUpgradeTestUtil.getFromVersion(), RollingUpgradeTestUtil.getToVersion())
-            .jgroupsProtocol("tcp")
+            .useCustomServerConfiguration("configuration/ClusteredServerTest.xml")
+            .jgroupsProtocol("test-tcp")
             .nodeCount(nodeCount);
       builder.handlers(
             uh -> {

--- a/server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java
+++ b/server/testdriver/core/src/main/java/org/infinispan/server/test/core/ContainerInfinispanServerDriver.java
@@ -84,7 +84,7 @@ public class ContainerInfinispanServerDriver extends AbstractInfinispanServerDri
    private static final String STARTUP_MESSAGE_REGEX = ".*ISPN080001.*";
    private static final String SHUTDOWN_MESSAGE_REGEX = ".*ISPN080003.*";
    private static final String CLUSTER_VIEW_REGEX = ".*ISPN000093.*(?<=\\()(%1$d)(?=\\)).*|.*ISPN000094.*(?<=\\()(%1$d)(?=\\)).*";
-   private static final int TIMEOUT_SECONDS = Integer.getInteger(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_TIMEOUT_SECONDS, 45);
+   private static final int TIMEOUT_SECONDS = Integer.getInteger(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_TIMEOUT_SECONDS, 50);
    private static final Long IMAGE_MEMORY = Long.getLong(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_MEMORY, null);
    private static final Long IMAGE_MEMORY_SWAP = Long.getLong(TestSystemPropertyNames.INFINISPAN_TEST_SERVER_CONTAINER_MEMORY_SWAP, null);
    public static final String INFINISPAN_SERVER_HOME = "/opt/infinispan";

